File: DKS-SPA/src/app/core/_models/s_f340-schedule.ts
import { Pagination } from "./pagination";
import { utilityConfig } from "../..//core/utility/utility-config";

export class SF340Schedule extends Pagination {
  dataType:string;
  factory: string;
  season: string;
  bpVer: string;
  cwaDateS : string;
  cwaDateE : string;
  /**
   *default set of searching parameters
   */
  constructor() {
    super();
    this.dataType = "DHO"; //預設DHO
    this.factory = utilityConfig.factory;//預設翔鴻程
    this.season = "";
    this.bpVer = "";
    this.cwaDateS ="";
    this.cwaDateE ="";
    this.isPaging = true; //開分頁
  }

  public setPagination(pagination: Pagination) {
    this.currentPage = pagination.currentPage;
    this.itemsPerPage = pagination.itemsPerPage;
    this.totalItems = pagination.totalItems;
    this.totalPages = pagination.totalPages;
  }
}

File: DKS-API/DTOs/SF340Schedule.cs
using System;
using DKS_API.Helpers;

namespace DKS_API.DTOs
{
    public class SF340Schedule : PaginationParams
    {
        public string dataType { get; set; }
        public string factory { get; set; }
        public string season { get; set; }
        public string bpVer { get; set; }
        public string cwaDateS { get; set; }
        public string cwaDateE { get; set; }
    }
}